### DISCUSSION TITLE
TOTP not working with some services

### DISCUSSION THREAD

**Discussion by 0xab3d** [dani-garcia/vaultwarden#3002] [Q&A] [answered]
### Subject of the issue

TOTP not working with some services such as _sucuri.net_

### Deployment environment

* vaultwarden version: 1.26

* Install method: Docker

* Clients used:  web vault and Edge extension

* Reverse proxy and version:  Sucuri SaaS

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:

### Steps to reproduce

Edit the Sucuri entry, try to add the TOTP seed, nothing happens. Once I click save, the Edge extension just logs out. Replicated this on the web as well.

### Expected behaviour

TOTP seed starts providing the 6 digit codes.
### Actual behaviour

The Edge extension and the web session logs out when I save the changes.

**Comment by BlackDex** [comment]
I'm not sure how Sucuri SaaS works. Does that add an extra layer of authentication, then that could be an issue.

**Comment by 0xab3d** [comment]
Hey, it is a reverse proxy (acting as WAF)

**Comment by BlackDex** [comment, marked as answer]
I think the WAF is removing some headers or other items, which breaks either the client or server.

Check the logs, maybe set the log-level to debug for some more info.

**Comment by 0xab3d** [comment]
You're totally right. I have checked the logs and turned out the WAF is blocking PUT methods..

**Reply by 0xab3d** [reply]
@BlackDex looks like I spoke too soon. TOTP is working however for some reason when I sync the db all new changes (after implementing the WAF) are disappearing. Any thoughts? The logs are not showing anything.

**Reply by 0xab3d** [reply]
I have reduced the caching on the WAF and it seems the issue is gone. Please let me if this makes sense to you, otherwise it could be a coincidence, right?

**Reply by tessus** [reply]
> I have reduced the caching on the WAF and it seems the issue is gone. 

This reminds me of the following: https://github.com/dani-garcia/vaultwarden/discussions/2984#discussioncomment-4400793

**Reply by 0xab3d** [reply]
@tessus thanks for sharing.

**Reply by BlackDex** [reply]
I'm not sure from the top of my head. But if you are able to disable caching on the json content type, i think you should be able to keep caching on, but only for the static files of course.

But i haven't tried something like that my self yet.
